src/common/commonutils/SshUtils.c

```diff
@@ -194,43 +284,44 @@ int CheckAppropriateCiphersForSsh(const char** ciphers, unsigned int numberOfCip
             {
                 TruncateAtFirst(value, ',');
 
-                for (j = 0; j < numberOfCiphers; j++)
-                {
-                    if (0 == strcmp(value, ciphers[j]))
-                    {
-                        cipherFound = true;
-                        break;
-                    }
-                }
-
-                if (false == cipherFound)
+                if (NULL == strstr(ciphers, value))
                 {
                     status = ENOENT;
                     OsConfigLogError(log, "CheckAppropriateCiphersForSsh: unapproved cipher '%s' found in SSH Server response", value);
                     OsConfigCaptureReason(reason, "Unapproved cipher '%s' found in SSH Server response", "%s, also cipher '%s' is unapproved", value);
                 }
 
                 i += strlen(value);
-
-                cipherFound = false;
-
                 FREE_MEMORY(value);
-
                 continue;
             }
         }
 
-        for (j = 0; j < numberOfCiphers; j++)
+        ciphersValueLength = strlen(ciphers);
+
+        // Check that all required ciphers are configured
+        for (i = 0; i < ciphersValueLength; i++)
         {
-            if (NULL == strstr(ciphersValue, ciphers[j]))
+            if (NULL == (value = DuplicateString(&(ciphers[i]))))
             {
-                status = ENOENT;
-                OsConfigLogError(log, "CheckAppropriateCiphersForSsh: required cipher '%s' not found in SSH Server response", ciphers[j]);
-                OsConfigCaptureReason(reason, "Required cipher '%s' not found in SSH Server response", "%s, also required cipher '%s' is not found", ciphers[j]);
+                OsConfigLogError(log, "CheckAppropriateCiphersForSsh: failed to duplicate ciphers string");
+                status = ENOMEM;
+                break;
             }
             else
             {
-                OsConfigLogInfo(log, "CheckAppropriateCiphersForSsh: required cipher '%s' found in SSH Server response", ciphers[j]);
+                TruncateAtFirst(value, ',');
+
+                if (NULL == strstr(ciphersValue, value))
+                {
+                    status = ENOENT;
+                    OsConfigLogError(log, "CheckAppropriateCiphersForSsh: required cipher '%s' not found in SSH Server response", &(ciphers[i]));
+                    OsConfigCaptureReason(reason, "Required cipher '%s' not found in SSH Server response", "%s, also required cipher '%s' is not found", &(ciphers[i]));
+                }
+
+                i += strlen(value);
+                FREE_MEMORY(value);
+                continue;
```

Comment:
nit: Seems redundant with `continue` here?

Reply:
It's nice to be extra explicit sometimes. Look at this like it were a comment.